src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -2026,6 +2026,17 @@ void GraphBuilder::invoke(Bytecodes::Code code) {
     code = Bytecodes::_invokespecial;
   }
 
+  // check if we need to blackhole the method
+  if (target->is_loaded() && target->return_type()->basic_type() == T_VOID &&
+      compilation()->directive()->should_blackhole(target)) {
+    if (try_blackhole(target)) {
```

Comment:
Since `try_blackhole` can't fail, why do the check at all?

Reply:
Right. This used to be modeled after `try_inline`, which could bail. `try_blackhole` cannot bail, so there is no reason to assert it. Fixed.